src/Lean/Meta/Coe.lean

```diff
@@ -32,26 +32,18 @@ partial def expandCoe (e : Expr) : MetaM Expr :=
             return .visit e.headBeta
       return .continue
 
-register_builtin_option maxCoeSize : Nat := {
```

Comment:
I'm so done with these arbitrary limits.  I spent an hour yesterday trying to figure out what broke with the latest update.  Why CoeHTCT worked but CoeT didn't, etc.

There's no reason to have a limit here, and I'm not willing to waste any more time debugging the random breakage this causes.